### PR TITLE
NAPSSPO-1024 - tssc-containers - tssc user doesn't actually exist, causes issues for rootless buildah

### PR DESCRIPTION
**Pull request by itewk**
## What is this PR About?
* tssc-base update to actually create tssc user
* tssc-base add args:
  - ARG TSSC_USER_NAME=tssc
  - ARG TSSC_USER_UID=1001
  - ARG TSSC_USER_GID=0
  - ARG TSSC_HOME_DIR=/home/tssc
* update all images to use TSSC_USER_UID
* remove depricated 'MAINTAINER" line from all Dockerfiles
* tssc-tool-maven - remove settings.xml

Related:
* NAPSSPO-1025 - tssc-jenkins-library - openscap container is running as privilaged
* NAPSSPO-1018 - tssc-jenkins-library - buildah container is running as privilaged

## effects
* https://github.com/rhtconsulting/tssc-python-package/pull/85
* https://github.com/rhtconsulting/tssc-jenkins-library/pull/11

## integration test:
* https://test-jenkins-unprivilaged.apps.tssc.rht-set.com/job/tssc-references/job/tssc-reference-app-quarkus-rest-json/job/feature%252FNAPSSPO-1018/11/console

### CODE REVIEW

File: tssc-base/Dockerfile.ubi8 [hunk 1]
```diff
@@ -47,12 +54,12 @@ RUN INSTALL_PKGS="gettext git rsync tar unzip which zip bzip2 python36 python3-p
 RUN alternatives --set python /usr/bin/python3 && \
     python -m pip install --upgrade pip
 
-# Configure home
-RUN mkdir -p /home/tssc && \
-    chown -R 1001:0 /home/tssc && \
-    chmod -R g+w /home/tssc 
+# Configure tssc user
+RUN useradd tssc --uid $TSSC_USER_UID --gid $TSSC_USER_GID --home-dir ${TSSC_HOME_DIR} --create-home --shell /sbin/noloing && \
```
Comment (kharyam) [inline tssc-base/Dockerfile.ubi8:58]: `/sbin/noloing` -> `/sbin/nologin`

**Review by kharyam — CHANGES_REQUESTED** [review]
Minor typo

File: tssc-base/Dockerfile.ubi8 (hunk 1 above)
Comment (itewk) [reply tssc-base/Dockerfile.ubi8]: fixed.

**Review by kharyam — APPROVED** [review]
Looks good

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -750,12 +750,12 @@ jobs:
 
       - name: Test Image 🧪
         run: |
-          echo "Running: docker run ${{ env.IMAGE_TAG_LOCAL }} java -version"
-          docker run -u 1001 ${{ env.IMAGE_TAG_LOCAL }} java -version
-          echo "Running: docker run ${{ env.IMAGE_TAG_LOCAL }} mvn --version"
-          docker run -u 1001 ${{ env.IMAGE_TAG_LOCAL }} mvn --version
-          echo "Running: docker run ${{ env.IMAGE_TAG_LOCAL }} -u 1001 touch ~/tssc/.m2/test.txt"
-          docker run -u 1001 ${{ env.IMAGE_TAG_LOCAL }} touch /home/tssc/.m2/test.txt
+          echo "Verify java installed"
```
Comment (kfrankli) [inline .github/workflows/publish.yaml:753]: Huh, I like this human language a lot more!

**Review by kfrankli — APPROVED** [review]
Really like the switch to human language for logs

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (itewk) [reply .github/workflows/publish.yaml:753]: glad to hear it :)

I didn't update all of them. but someone else can do that :).

**Comment by itewk** [comment]
auto publish of images: https://github.com/rhtconsulting/tssc-containers/actions/runs/270736709
manual push images:
```
14:26 $ ./manual-publish.sh edge tssc quay.io

--------------------

LOGIN: quay.io
Authenticating with existing credentials...
Existing credentials are valid. Already logged in to quay.io

--------------------

CALCULATE TAGS

--------------------

CONFIRMATION
    IMAGES: tssc-tool-buildah tssc-tool-openscap
    IMAGE VERSION: edge
    IMAGE TAGS: edge sha-ef35f8e3
    REGISTRY REPOSITORY: tssc
    REGISTRY_URI: quay.io

Are you sure you want to build, tag, and push? (Y/N): Y
```

**Pull request merged**